lib/pay/env.rb

```diff
@@ -40,11 +38,15 @@ def env
     end
 
     def secrets
-      Rails.application.secrets
+      is_rails_version_below_7_2? ? Rails.application.secrets : nil
```

Comment:
```suggestion
      Rails.application.secrets if Rails.application.respond_to?(:secrets)
```
Does this work instead? Then it's consistent with `credentials` and we don't need any code to check the specific Rails version.

Reply:
I guess this would still print the deprecation warning in Rails 7.1, but it would handle Rails 7.2 safely.